### ISSUE TITLE
dot-notation rule complains on Set.delete()

### ISSUE THREAD

**Issue by mockdeep** [eslint/eslint#7457]
* **ESLint Version:** 3.8.1

The following code causes the `dot-notation` rule to complain for me:

```js
mySet.delete(myItem);
```

It tells me `error  .delete is a syntax error  dot-notation`. `delete` is a valid method on `Set`, so it seems like this shouldn't cause an error.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/delete

**Comment by not-an-aardvark** [comment]
I think this is happening because you have [`allowKeywords`](http://eslint.org/docs/rules/dot-notation#options) set to `false` in your config, which requires bracket notation for all property access if the property happens to be a keyword, like `delete`.

To fix it, change your config file to include something like this:

```json
{
  "rules": {
    "dot-notation": ["error", { "allowKeywords": true }]
  }
}
```

**Comment by mockdeep** [comment]
It seems a little strange to have to `allowKeywords` for something like this, though, being that this particular keyword is also implemented as a method on a core JS class. It seems like more of a special case when it comes to this rule...

**Comment by not-an-aardvark** [comment]
I think `allowKeywords` is true by default, so you can also just use:

```json
{
  "rules": {
    "dot-notation": "error"
  }
}
```

I think the reasoning for the `allowKeywords` option is because in some older JS engines, something like `foo.delete(bar)` is actually a SyntaxError because it parses `delete` as a keywords.

**Comment by platinumazure** [comment]
Confirmed that `allowKeywords` is true by default.

I'm going to close this issue as the question seems to have been answered. Please feel free to continue the discussion here if you have further questions.
